### PR TITLE
Add mention of `rlib/setup-renv` in CI article

### LINKED ISSUE #944: Update doc to mention r-lib/actions/setup-renv ?
Currently, doc about CI does not mention r-lib/actions/setup-renv
https://rstudio.github.io/renv/articles/ci.html#github-actions
and it points also to https://github.com/actions/cache/blob/main/examples.md#r---renv as an example of using cache with renv. 

Maybe https://github.com/r-lib/actions/tree/v2-branch/setup-renv should be mentioned as it is one step to 
* install renv 
* setup the cache 
* restore the package based on the lockfile

this is the recommended way right ?

### PR DESCRIPTION
**Pull request by cderv**
Closes #944 

This rewrites the part about CI with github action to point and promote usage of `r-lib/actions/setup-renv`

### CODE REVIEW

File: vignettes/ci.Rmd [hunk 1]
```diff
@@ -40,14 +40,21 @@ providing an alternate R package repository to use during restore below.
 
 ## GitHub Actions
 
-When using GitHub Actions, you typically need two steps:
+Using `renv` in a Github actions workflow can be done in two ways:
```
Comment (kevinushey) [inline vignettes/ci.Rmd:43]:
```suggestion
Here, we describe two common approaches for integrating `renv` with a [GitHub Actions](https://github.com/features/actions) workflow.
```

File: vignettes/ci.Rmd [hunk 2]
```diff
@@ -40,14 +40,21 @@ providing an alternate R package repository to use during restore below.
 
 ## GitHub Actions
 
-When using GitHub Actions, you typically need two steps:
+Using `renv` in a Github actions workflow can be done in two ways: 
 
-1. Cache any packages installed by `renv`,
-2. Use `renv::restore()` to restore packages.
+* Using directly the cache mechanism of Github action and running `renv` functions,
```
Comment (kevinushey) [inline vignettes/ci.Rmd:45]:
```suggestion
* Use GitHub's built-in cache action together with existing `renv` functionality;
```

File: vignettes/ci.Rmd [hunk 3]
```diff
@@ -40,14 +40,21 @@ providing an alternate R package repository to use during restore below.
...
+* Using an helper action from `r-lib` organization that will do everything for you
```
Comment (kevinushey) [inline vignettes/ci.Rmd:46]:
```suggestion
* Use the `setup-renv` action maintained by the `r-lib` organization.
```

File: vignettes/ci.Rmd [hunk 4]
```diff
@@ -40,14 +40,21 @@ providing an alternate R package repository to use during restore below.
...
+
+### Using the Github actions cache with renv
```
Comment (kevinushey) [inline vignettes/ci.Rmd:48]:
```suggestion
### Using the GitHub Actions Cache with renv
```

File: vignettes/ci.Rmd [hunk 5]
```diff
@@ -40,14 +40,21 @@ providing an alternate R package repository to use during restore below.
...
+
+To benefit from `renv` value in a Github action, you typically need two steps:
```
Comment (kevinushey) [inline vignettes/ci.Rmd:50]:
```suggestion
When using `renv` in your own custom GitHub action workflow, there are two main requirements:
```

File: vignettes/ci.Rmd [hunk 6]
```diff
@@ -70,6 +77,27 @@ steps:
 
 See also the [example][github-actions-renv] on GitHub actions.
 
+### Simplify using a extenral helper Github action
```
Comment (kevinushey) [inline vignettes/ci.Rmd:80]:
```suggestion
### Using r-lib/actions/setup-renv
```

File: vignettes/ci.Rmd [hunk 7]
```diff
@@ -70,6 +77,27 @@ steps:
...
+
+`r-lib` organization offers some actions for R users, and among them a [`setup-renv`][r-lib-actions-renv] action. To use this action, just add a `uses` step in your workflow:
```
Comment (kevinushey) [inline vignettes/ci.Rmd:82]:
```suggestion
The `r-lib` organization offers some actions for R users, and among them a [`setup-renv`][r-lib-actions-renv] action is provided for projects using `renv`. To use this action, you can add the following steps to your workflow: 
```

File: vignettes/ci.Rmd [hunk 8]
```diff
@@ -70,6 +77,27 @@ steps:
...
+
+`r-lib` organization offers some actions for R users, and among them a [`setup-renv`][r-lib-actions-renv] action. To use this action, just add a `uses` step in your workflow: 
+
+```yaml
+steps:
+- uses: actions/checkout@v3
+- uses: r-lib/actions/setup-r@v2
+- uses: r-lib/actions/setup-renv@v2
+  with:
+    profile: '"shiny"'
```
Comment (kevinushey) [inline vignettes/ci.Rmd:90]: Is this necessary?

File: vignettes/ci.Rmd [hunk 9]
```diff
@@ -70,6 +77,27 @@ steps:
...
+```
+
+The `r-lib/actions/setup-renv` will do all that is needed to befenit from `renv` value in a Github action workflow:
```
Comment (kevinushey) [inline vignettes/ci.Rmd:93]:
```suggestion
Using these steps, the following actions will then be performed in your workflow:
```

File: vignettes/ci.Rmd [hunk 10]
```diff
@@ -70,6 +77,27 @@ steps:
...
+- uses: actions/checkout@v3
+- uses: r-lib/actions/setup-r@v2
+- uses: r-lib/actions/setup-renv@v2
+  with:
+    profile: '"shiny"'
+```
+
+The `r-lib/actions/setup-renv` will do all that is needed to befenit from `renv` value in a Github action workflow: 
+
+* Install `renv` package
```
Comment (kevinushey) [inline vignettes/ci.Rmd:95]:
```suggestion
* `renv` will be installed, via `install.packages("renv")`,
```

File: vignettes/ci.Rmd [hunk 11]
```diff
@@ -70,6 +77,27 @@ steps:
...
+* Setting up the cache
```
Comment (kevinushey) [inline vignettes/ci.Rmd:96]:
```suggestion
* `renv` will be configured to use the GitHub cache,
```

File: vignettes/ci.Rmd [hunk 12]
```diff
@@ -70,6 +77,27 @@ steps:
...
+- uses: r-lib/actions/setup-renv@v2
+  with:
+    profile: '"shiny"'
+```
+
+The `r-lib/actions/setup-renv` will do all that is needed to befenit from `renv` value in a Github action workflow: 
+
+* Install `renv` package
+* Setting up the cache 
+* Activate the right `renv` profile
```
Comment (kevinushey) [inline vignettes/ci.Rmd:97]:
```suggestion
* If provided via a `with: profile:` key, that `renv` profile will be activated,
```

File: vignettes/ci.Rmd [hunk 13]
```diff
@@ -70,6 +77,27 @@ steps:
...
+* Restore the environment with `renv::restore()`
```
Comment (kevinushey) [inline vignettes/ci.Rmd:98]:
```suggestion
* The project will be restored via `renv::restore()`.
```

File: vignettes/ci.Rmd [hunk 14]
```diff
@@ -70,6 +77,27 @@ steps:
...
+
+Unless you need more control, this actions is all you need for using `renv` with Github action.
```
Comment (kevinushey) [inline vignettes/ci.Rmd:100]:
```suggestion
After this, any steps using R will use the active `renv` project by default.
```

File: vignettes/ci.Rmd (hunk 8 above)
Comment (cderv) [reply vignettes/ci.Rmd:90]: No it is an example of the fact you can use profile feature. But let's leave that for the `setup-renv` documentation.

Comment (cderv) [inline vignettes/ci.Rmd:90]:
```suggestion
- uses: r-lib/actions/setup-renv@v2
```

**Comment by cderv** [comment]
Thanks a lot for all the suggestion. I agree with them all ! You're obviously more at ease with the English language 😉

**Review by kevinushey — APPROVED** [review]
Thanks!

**Pull request merged**